### PR TITLE
Fix problems with Firebase 5.x

### PR DESCRIPTION
**Pull request by Narayane**
Firebase AuthDataResultCallback does not return directly a user anymore, update code to fix it

### CODE REVIEW

File: Sources/Auth+Rx.swift [hunk 1]
```diff
@@ -43,7 +43,15 @@ extension Reactive where Base: Auth {
 extension Reactive where Base: Auth {
     public func createUser(withEmail email: String, password: String) -> Single<User> {
```
Comment (sgr-ksmt) [inline Sources/Auth+Rx.swift]: It should change return type from `Single<User>` to `Single<AuthDataResult>`.
By doing so, it looks like the following

```swift
    public func createUser(withEmail email: String, password: String) -> Single<AuthDataResult> {
        return .create { observer in
            self.base.createUser(withEmail: email, password: password, completion: singleEventHandler(observer))
            return Disposables.create()
        }
    }

```

`signInAnonymously`, `signIn(withEmail:password:)` and `signInWith(customToken:` also should change as same.

**Review by sgr-ksmt — CHANGES_REQUESTED** [review]
Thansk for submitting PR.
Please change code a little 🙏

**Comment by Narayane** [comment]
Done :)

**Review by sgr-ksmt — APPROVED** [review]
Thanks 😄

**Comment by sgr-ksmt** [comment]
I will submit new version to Cocoapods. Please wait a while.

**Comment by Narayane** [comment]
Perfect, thanks :)

**Pull request merged**